kit/src/components/embeds/youtube/mod.rs

```diff
@@ -7,16 +7,39 @@ pub struct Props {
 
 #[allow(non_snake_case)]
 pub fn YouTubePlayer(cx: Scope<Props>) -> Element {
-    cx.render(
-        rsx!(
-            div {
-                id: "youtube-player",
-                iframe {
-                    src: "{cx.props.video_url}",
-                    allow: "accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture",
-                    allowfullscreen: true,
-                },
-            }
-        )
-    )
+    let src_video = match extract_video_id_from_embed_url(&cx.props.video_url) {
+        Some(video_id) => format!("https://www.youtube.com/embed/{}", video_id),
+        None => cx.props.video_url.clone(),
+    };
+
+    cx.render(rsx!(
+        div {
+            id: "youtube-player",
+            iframe {
+                src: "{src_video}",
+                frame_border: "0",
+                allow: "accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture",
+                allowfullscreen: true,
+            },
+        }
+    ))
+}
+
+fn extract_video_id_from_embed_url(embed_url: &str) -> Option<String> {
+    let base_embed_url = "https://www.youtube.com/embed/";
+
+    // Using strip_prefix to remove the base URL if it exists
+    if let Some(video_id_with_params) = embed_url.strip_prefix(base_embed_url) {
+        // Now, extract the video ID before any additional parameters
+        let video_id = match video_id_with_params.find('&') {
+            Some(end_pos) => &video_id_with_params[..end_pos], // Extract the video ID up to the start of additional parameters
+            None => video_id_with_params, // If there is no '&', the entire string is the video ID
+        };
+
+        // Return the video ID
+        return Some(video_id.to_string());
+    }
+
+    // Return None if the URL doesn't start with the expected base embed URL
+    None
```

Comment:
You can try an `Option` if a function returns an `Option`

```suggestion
fn extract_video_id_from_embed_url(embed_url: &str) -> Option<String> {
    let base_embed_url = "https://www.youtube.com/embed/";

    // Using strip_prefix to remove the base URL if it exists
    let video_id_with_params = embed_url.strip_prefix(base_embed_url)?;

    // Now, extract the video ID before any additional parameters
    let video_id = match video_id_with_params.find('&') {
        Some(end_pos) => &video_id_with_params[..end_pos], // Extract the video ID up to the start of additional parameters
        None => video_id_with_params, // If there is no '&', the entire string is the video ID
    };

    // Return the video ID
    Some(video_id.to_string())
```

Reply:
Thanks, did this change!